pkg/oam/util/helper.go

```diff
@@ -196,39 +200,53 @@ func FetchWorkloadDefinition(ctx context.Context, r client.Reader, dm discoverym
 	if err != nil {
 		return nil, err
 	}
-	// Fetch the corresponding workloadDefinition CR
-	workloadDefinition, err := GetWorkloadDefinition(ctx, r, wldName)
-	if err != nil {
+	workloadDefinition := new(v1alpha2.WorkloadDefinition)
+	if err = GetDefinition(ctx, r, workloadDefinition, wldName); err != nil {
 		return nil, err
 	}
 	return workloadDefinition, nil
 }
 
-// GenNamespacedDefinitionName generate definition name with customized namespace
-func GenNamespacedDefinitionName(dn string) types.NamespacedName {
-	if dns := os.Getenv(DefinitionNamespaceEnv); dns != "" {
-		return types.NamespacedName{Name: dn, Namespace: dns}
+// GetDefinitionNamespaceWithCtx gen sysNs and appNs from ctx
```

Comment:
```suggestion
// GetDefinitionNamespaceWithCtx will get namespace from context, it will try get `appDefinitionNamespace` key, if not found, will use default system level namespace defined in `systemvar.SystemDefinitonNamespace`
```

you should explain more detail for this function.